lib/model/budget.dart

```diff
@@ -104,20 +125,17 @@ class BudgetMethods extends SossoldiDatabase {
     // You can use `rawUpdate` to write the query in SQL
     return database.update(
       budgetTable,
-      item.toJson(),
-      where:
-      '${BudgetFields.id} = ?',
+      item.toJson(update: true),
+      where: '${BudgetFields.id} = ?',
       whereArgs: [item.id],
     );
   }
 
   Future<int> deleteById(int id) async {
     final database = await SossoldiDatabase.instance.database;
 
-    return await database.delete(budgetTable,
-        where:
-        '${BudgetFields.id} = ?',
-        whereArgs: [id]);
+    // logical delete by setting active to 0
+    return await database
+        .rawUpdate('UPDATE Budget SET active = 0 WHERE id = ?', [id]);
```

Comment:
Not sure if we want this behavior here. I think this delete function should actually delete the record on db, and when the user removes a budget, our BE logic should use `updateItem()` function to change the `active` field to 0.